Vaadin 8 version

Artur- opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/context-menu/35)
<!-- Reviewable:end -->

hesara commented:
Reviewed 5 of 8 files at r1.
Review status: 5 of 8 files reviewed at latest revision, 4 unresolved discussions, some commit checks broke.

---

*[vaadin-context-menu-addon/pom.xml, line 15 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRYev0u8ZQuMSI-aBHM:-KRYev0u8ZQuMSI-aBHN:b-x1kfzj) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/pom.xml#L15)):*
> ```XML
>         <maven.compiler.target>1.8</maven.compiler.target>
>         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
>         <vaadin.version>8.0.0.alpha2</vaadin.version>
> ```

(anticipating a bit)

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/client/MyVMenuBar.java, line 39 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRYfT0_Kb-BrRnN4zlL:-KRYfT0a7vPwnU10oawL:b-dzf33) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/client/MyVMenuBar.java#L39)):*
> ```Java
>     // this method has a couple lines added, marked with FIXME
>     @Override
>     public boolean handleNavigation(int keycode, boolean ctrl, boolean shift) {
> ```

we should really get rid of this copy of much of VMenuBar code by making the necessary bits of framework code extensible

---

*[vaadin-context-menu-demo/pom.xml, line 13 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRYfYqdZM5xKdlQQWgl:-KRYfYqdZM5xKdlQQWgm:bajxi0v) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/pom.xml#L13)):*
> ```XML
>     <properties>
>         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
>         <vaadin.version>8.0.0.alpha2</vaadin.version>
> ```

(anticipating)

---

*[vaadin-context-menu-demo/src/main/resources/com/vaadin/addon/contextmenu/demo/DemoWidgetSet.gwt.xml, line 37 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRYfm8Rm_78pA2B1dAs:-KRYfm8Rm_78pA2B1dAt:b-sdl7eh) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/resources/com/vaadin/addon/contextmenu/demo/DemoWidgetSet.gwt.xml#L37)):*
> ```XML
>     
> 
>     <inherits name="com.vaadin.DefaultWidgetSet" />
> ```

(technically not needed, but I guess this gets added automatically and doesn't hurt)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: 5 of 8 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/client/MyVMenuBar.java, line 39 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRYfT0_Kb-BrRnN4zlL:-KR_-9NhxNFHxGhpGF-w:b-bdey4o) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/client/MyVMenuBar.java#L39)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary>
> we should really get rid of this copy of much of VMenuBar code by making the necessary bits of framework code extensible
</details>
Yes, not needed for the first Vaadin 8 compatible version though

---

*[vaadin-context-menu-demo/src/main/resources/com/vaadin/addon/contextmenu/demo/DemoWidgetSet.gwt.xml, line 37 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRYfm8Rm_78pA2B1dAs:-KR_-FRga576AG1uM4Xi:bklzvay) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/resources/com/vaadin/addon/contextmenu/demo/DemoWidgetSet.gwt.xml#L37)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary>
> (technically not needed, but I guess this gets added automatically and doesn't hurt)
</details>
Doesn't matter, it's a Framework 8 bug

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 2 of 8 files at r1.
Review status: 7 of 8 files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[vaadin-context-menu-addon/pom.xml, line 85 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRb8sxzsrGkMVe8wWdF:-KRb8sxzsrGkMVe8wWdG:b4ucz8y) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/pom.xml#L85)):*
> ```XML
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>vaadin-compatibility-server</artifactId>
> ```

now context menu add-on forces the use of compatibility packages and the bigger compatibility widgetset - what is the plan to get rid of this? split the add-on later?

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/client/MyVMenuBar.java, line 39 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRYfT0_Kb-BrRnN4zlL:-KRb85ycP_pBVMLUgyCz:b-ny9wuh) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/client/MyVMenuBar.java#L39)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> Yes, not needed for the first Vaadin 8 compatible version though
</details>
can be addressed later

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 1 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRb8W0D7mMxFBSiPjiX:-KRb8W0EuKHph5tgYSh2:b-mhuhs3) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L1)):*
> ```Java
> package com.vaadin.v7.addon.contextmenu;
> ```

is the v7 a top-level thing, or within the add-on?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks broke.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/GridContextMenu.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLAbD2W5W4lzph1y0:-KRbLAbD2W5W4lzph1y1:b-vi30yr) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/GridContextMenu.java#L22)):*
> ```Java
>     private void addGridSectionContextMenuListener(final Section section,
>             final GridContextMenuOpenListener<T> listener) {
>         addContextMenuOpenListener(new ContextMenuOpenListener() {
> ```

May be extract to lambda-method ?

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLQMdWrsqYxJYr1S0:-KRbLQMdWrsqYxJYr1S1:b-9ffge0) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L22)):*
> ```Java
>     private void addGridSectionContextMenuListener(final Section section,
>             final GridContextMenuOpenListener listener) {
>         addContextMenuOpenListener(new ContextMenuOpenListener() {
> ```

Extract to lambda-method.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 58 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLdlHpBgfYB2vs-VR:-KRbLdlHpBgfYB2vs-VS:buv2z6g) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L58)):*
> ```Java
>         public void onContextMenuOpen(GridContextMenuOpenEvent event);
> 
>         public static class GridContextMenuOpenEvent
> ```

Double nested class ?

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 86 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLui9O5KsaxKngOvM:-KRbLui9O5KsaxKngOvN:b77t6iu) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L86)):*
> ```Java
>         com.vaadin.v7.addon.contextmenu.GridContextMenu contextMenu2 = new com.vaadin.v7.addon.contextmenu.GridContextMenu(
>                 grid);
>         contextMenu2.addGridHeaderContextMenuListener(e -> {
> ```

Too long lambda.
Would be good to extract to a method.

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 94 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbM-AAni6WipXm0ZCV:-KRbM-AAni6WipXm0ZCW:b77t6iu) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L94)):*
> ```Java
> 
>         });
>         contextMenu2.addGridBodyContextMenuListener(e -> {
> ```

Too long lambda.
Would be good to extract to a method.

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 120 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbM2l-p6144FHaQ5uc:-KRbM2l-p6144FHaQ5ud:b-hu8n44) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L120)):*
> ```Java
>         grid.setDataSource(ds);
>         contextMenu2.addContextMenuOpenListener(e -> {
>             GridContextClickEvent<String[]> gridE = (GridContextClickEvent<String[]>) e
> ```

It's definitely too long lambda.

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 157 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbMDJJ013nJyN3ZcHE:-KRbMDJJ013nJyN3ZcHF:b77t6iu) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L157)):*
> ```Java
>                 Collections.singletonList(new String[] { "foo", "bar" }));
>         grid.setDataSource(dataSource);
>         gridContextMenu.addGridHeaderContextMenuListener(e -> {
> ```

Too long lambda.
Would be good to extract to a method.

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 165 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbMEdc4UeimiQns6Bv:-KRbMEdc4UeimiQns6Bw:b77t6iu) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L165)):*
> ```Java
>         });
> 
>         gridContextMenu.addGridBodyContextMenuListener(e -> {
> ```

Too long lambda.
Would be good to extract to a method.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks broke.

---

*[vaadin-context-menu-addon/pom.xml, line 85 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRb8sxzsrGkMVe8wWdF:-KRbZGxg9mN0rIINJf4H:b-k82jui) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/pom.xml#L85)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary>
> now context menu add-on forces the use of compatibility packages and the bigger compatibility widgetset - what is the plan to get rid of this? split the add-on later?
</details>
It uses `provided` scope so it should not force the compatibility packages upon anyone. Maybe the demo should be split into a "v7" part and a "v8" part to ensure that this actually works

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 1 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRb8W0D7mMxFBSiPjiX:-KRbZa2zOjWBlL6BC13K:bsrxjfr) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L1)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary>
> is the v7 a top-level thing, or within the add-on?
</details>
I would see it as a top level thing, to keep it the same everywhere

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

hesara commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks broke.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 1 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRb8W0D7mMxFBSiPjiX:-KRgxB51AeOOCYA0di4r:betr69j) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L1)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> I would see it as a top level thing, to keep it the same everywhere
</details>
good enough for me for now, might need to be coordinated between add-ons

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks broke.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLQMdWrsqYxJYr1S0:-KRbZdKTk0WuZYqCdvOQ:bw9ng8j) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L22)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> Extract to lambda-method.
</details>
This is just a rename of the existing version

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 58 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLdlHpBgfYB2vs-VR:-KRh2DZEoAAUoRbdT8ub:bw9ng8j) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L58)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> Double nested class ?
</details>
This is just a rename of the existing version

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 86 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLui9O5KsaxKngOvM:-KRh2VpYHfera-8xUIyz:b-gve2o6) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L86)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> Too long lambda.
> Would be good to extract to a method.
</details>
Refactored "demos"

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 94 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbM-AAni6WipXm0ZCV:-KRh2XiC0-YQjesfJhbr:b-gve2o6) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L94)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> Too long lambda.
> Would be good to extract to a method.
</details>
Refactored "demos"

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 120 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbM2l-p6144FHaQ5uc:-KRh2Z2huFobFqqh8eeE:b-gve2o6) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L120)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> It's definitely too long lambda.
</details>
Refactored "demos"

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 157 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbMDJJ013nJyN3ZcHE:-KRh2ZalQ4GHDfQd0aqn:b-gve2o6) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L157)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> Too long lambda.
> Would be good to extract to a method.
</details>
Refactored "demos"

---

*[vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java, line 165 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbMEdc4UeimiQns6Bv:-KRh2_-9Ez93rwbOCyGN:b-gve2o6) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-demo/src/main/java/com/vaadin/addon/contextmenu/ContextmenuUI.java#L165)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> Too long lambda.
> Would be good to extract to a method.
</details>
Refactored "demos"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 1 of 14 files at r2.
Review status: 5 of 18 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLQMdWrsqYxJYr1S0:-KRhU6olsTH79cqVFlmY:b-lxin0d) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L22)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> This is just a rename of the existing version
</details>
What do you mean ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 12 of 14 files at r2.
Review status: 17 of 18 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-context-menu-vaadin7-test/src/main/java/com/vaadin/v7/addon/contextmenu/test/ContextmenuUI.java, line 79 at r2](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRhWEuGdvs1zN-cE1mB:-KRhWEuGdvs1zN-cE1mC:b-d0wxhj) ([raw file](https://github.com/vaadin/context-menu/blob/36d5b85e066fe367d2edc9ed0fa2dba1f9f7cdbc/vaadin-context-menu-vaadin7-test/src/main/java/com/vaadin/v7/addon/contextmenu/test/ContextmenuUI.java#L79)):*
> ```Java
>     }
> 
> }
> ```

new line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: 17 of 18 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLQMdWrsqYxJYr1S0:-KRhWWZYxBAmSEyaDHvB:b23ymt1) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L22)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> What do you mean ?
</details>
It's `vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/GridContextMenu.java` from master

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: 16 of 18 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-context-menu-vaadin7-test/src/main/java/com/vaadin/v7/addon/contextmenu/test/ContextmenuUI.java, line 79 at r2](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRhWEuGdvs1zN-cE1mB:-KRhlyuRik9uMXHo-tOd:b-896fix) ([raw file](https://github.com/vaadin/context-menu/blob/36d5b85e066fe367d2edc9ed0fa2dba1f9f7cdbc/vaadin-context-menu-vaadin7-test/src/main/java/com/vaadin/v7/addon/contextmenu/test/ContextmenuUI.java#L79)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> new line
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 1 of 14 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLQMdWrsqYxJYr1S0:-KRi77ImRgeRGtU6fx5_:b-59rldi) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L22)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> It's `vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/GridContextMenu.java` from master
</details>
So no need to improve ?
It's up to you.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/context-menu/35#-KRbLQMdWrsqYxJYr1S0:-KRiAAw8_0URAYeZNpZX:b-20wsr8) ([raw file](https://github.com/vaadin/context-menu/blob/d51c4fb139e9dc758d31b1d09aa6f6ade5453243/vaadin-context-menu-addon/src/main/java/com/vaadin/v7/addon/contextmenu/GridContextMenu.java#L22)):*
<details><summary><i>Previously, denis-anisimov wrote…</i></summary>
> So no need to improve ?
> It's up to you.
</details>
wontfix, if there is ever a bugfix to the Vaadin 7 version, it's easier to port when the code is the same

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/context-menu/35)*
<!-- Sent from Reviewable.io -->
